server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -0,0 +1,139 @@
+            REMOTE_STORE_REPOSITORY_SETTINGS_ATTRIBUTE_KEY_PREFIX,
+            repositoryName
+        );
+        Map<String, String> settingsMap = node.getAttributes()
+            .keySet()
+            .stream()
+            .filter(key -> key.startsWith(settingsAttributeKeyPrefix))
+            .collect(Collectors.toMap(key -> key.replace(settingsAttributeKeyPrefix, ""), key -> validateAttributeNonNull(key)));
+
+        if (settingsMap.isEmpty()) {
+            throw new IllegalStateException(
+                "joining node [" + this.node + "] doesn't have settings attribute for [" + repositoryName + "] repository"
+            );
+        }
+
+        return settingsMap;
+    }
+
+    private RepositoryMetadata buildRepositoryMetadata(String name) {
+        String type = validateAttributeNonNull(String.format(Locale.getDefault(), REMOTE_STORE_REPOSITORY_TYPE_ATTRIBUTE_KEY_FORMAT, name));
+        Map<String, String> settingsMap = validateSettingsAttributesNonNull(name);
+
+        Settings.Builder settings = Settings.builder();
+        settingsMap.forEach(settings::put);
+
+        // Repository metadata built here will always be for a system repository.
+        settings.put(BlobStoreRepository.SYSTEM_REPOSITORY_SETTING.getKey(), true);
+
+        return new RepositoryMetadata(name, type, settings.build());
+    }
+
+    private RepositoriesMetadata buildRepositoriesMetadata() {
+        List<RepositoryMetadata> repositoryMetadataList = new ArrayList<>();
+        Set<String> repositoryNames = new HashSet<>();
+
+        repositoryNames.add(validateAttributeNonNull(REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY));
+        repositoryNames.add(validateAttributeNonNull(REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY));
+
+        for (String repositoryName : repositoryNames) {
+            repositoryMetadataList.add(buildRepositoryMetadata(repositoryName));
+        }
+
+        return new RepositoriesMetadata(repositoryMetadataList);
+    }
+
+    public static boolean isRemoteStoreAttributePresent(Settings settings) {
+        return settings.getByPrefix(Node.NODE_ATTRIBUTES.getKey() + REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX).isEmpty() == false;
+    }
+
+    public RepositoriesMetadata getRepositoriesMetadata() {
+        return this.repositoriesMetadata;
+    }
+
+    @Override
+    public int hashCode() {
+        // We will hash the id and repositories metadata as its highly unlikely that two nodes will have same id and
+        // repositories metadata but are actually different.
+        return Objects.hash(node.getEphemeralId(), repositoriesMetadata);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        RemoteStoreNodeAttribute that = (RemoteStoreNodeAttribute) o;
+
+        return this.getRepositoriesMetadata().equalsIgnoreGenerations(that.getRepositoriesMetadata());
+    }
```

Comment:
The equals and hashCode methods are not consistent. equals method would be true as long as Repos metadata are same but hashCode uses node EphemeralId as well. So across nodes or for the same node if it is bounced hashCode would return different value. hashCode must be same for 2 objects if o1.equals(o2) == true